app/session.go

```diff
@@ -144,6 +173,15 @@ func (a *App) ClearSessionCacheForUserSkipClusterSend(userId string) {
 	a.InvalidateWebConnSessionCacheForUser(userId)
 }
 
+func (a *App) clearSessionCacheForUserSet(users map[string]struct{}) {
+	// if this ever gets reused, it would make sense to change
```

Comment:
Instead of clearing the cache sequentially for each individual user, could you add support to the cache for flushing it in it's entirety? Doing it like this is going to generate a DoS on the HA cluster cluster messages being triggered each iteration of the loop below.

Reply:
So the cluster part is basically a cache? I'll take a better look at it as I think I had the wrong concept about it

Reply:
The cluster is made up of a number of instances of the Mattermost server all connected to the same underlying SQL database. However, since these caches are in memory, they need to be synchronized across all the Mattermost servers through these cluster messages. My suggestion is that since we are removing *all* the sessions in one go, you can have one cluster message tell the other Mattermost servers to also flush the whole of their session caches, rather than a message per session that is removed.

Reply:
If you take a look in `app/cluster_handlers.go` you can see how the different cluster message handlers are registered. I would suggest creating a new one similar to `CLUSTER_EVENT_CLEAR_SESSION_CACHE_FOR_USER` but for flushing the entire session cache instead of just clearing it for one user.